web/modules/custom/solution/src/Plugin/Validation/Constraint/UniqueSolutionTitleValidator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\solution\Plugin\Validation\Constraint;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\rdf_entity\Entity\Rdf;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates that a field is unique for the given entity type within a bundle.
+ */
+class UniqueSolutionTitleValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($items, Constraint $constraint) {
+    if (!$item = $items->first()) {
+      return;
+    }
+    $field_name = $items->getFieldDefinition()->getName();
+    /** @var \Drupal\rdf_entity\RdfInterface $entity */
+    $entity = $items->getEntity();
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $id_key = $entity->getEntityType()->getKey('id');
+
+    // If the is_version_of field is not empty, check for the same title.
+    if (!empty($entity->get('field_is_is_version_of')
+      ->getValue()[0]['target_id'])
+    ) {
+      /** @var \Drupal\rdf_entity\RdfInterface $parent */
+      $parent = Rdf::load($entity->get('field_is_is_version_of')
+        ->getValue()[0]['target_id']);
+
+      if ($parent->label() == $entity->label()) {
+        // The release has the same name as the solution.
+        return;
```

Comment:
Also keep in mind that this validation file was built with the specifications that existed when the ticket was implemented. The functionality might need to change but according to the data I had at that moment, this was the constraint that had to be applied (now that I actually describe it in the comments :/ )